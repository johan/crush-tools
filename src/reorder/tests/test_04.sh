test_number=04
description="single input file, label-specified swap/move"

subtests=('-m Field-2,Field-0' '-m Field-2,1' '-m 3,Field-0'\
          '-s Field-1,Field-0' '-s Field-2,1' '-s 1,Field-2') 
subtest_desc=('move: label,label' 'move: label,index' 'move: index,label'\
              'swap: label,label' 'swap: label,index' 'swap: index,label')

for i in `seq 0 $(( ${#subtests[*]} - 1 ))`; do
  $bin -d , ${subtests[$i]} $test_dir/test_labeled.in \
    > $test_dir/test_$test_number.$i.out 2>&1
  if [ $? -ne 0 ] ||
     [ "`diff -q $test_dir/test_$test_number.$i.expected \
              $test_dir/test_$test_number.$i.out`" ]; then
    test_status $test_number $i "$description (${subtest_desc[$i]})" FAIL
  else
    rm $test_dir/test_$test_number.$i.out
    test_status $test_number $i "$description (${subtest_desc[$i]})" PASS
  fi
done
