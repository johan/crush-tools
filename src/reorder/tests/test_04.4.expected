Field-2,Field-1,Field-0
21,11,01
22,12,02
